fernando-gomez/precourse/airlines/airlines.js

```diff
@@ -0,0 +1,146 @@
+
+//Pushes last 5 flights into an array
+function lastFlightsArray(){
+
+        for(let i=flights.length; i > (flights.length - 6); i--){
+
+            lastFlights.push(flights[i]);
+        }
+        // Deletes first element (null) of lastFlights string
+        lastFlights.shift();
+}
+//Pushes flights with scales into an array
+function scaleFlightsArray(){
+
+    for(let i in flights){
+
+        if (flights[i].scale == true){
+            scaleFlights.push(flights[i]);         
+        }
+    }
+
+    scaleFlights.shift();   
+
+}
+// Calculates average price of all flights
+function flightsAveragePrice(){
+
+    for (let i in flights){
+        averagePrice += flights[i].cost;
+    }
+    averagePrice /= flights.length;
+}
+
+/************ Arrays showing ************/
+
+//Shows all flights for today
+function showFlights(){
+
+    for(let i in flights){
+        // Changes console log depending on scales number
+        if (flights[i].scale == false){
+            console.log(`${flights[i].id} - El vuelo con origen ${flights[i].from} y destino ${flights[i].to} tiene un coste de ${flights[i].cost}€ y no realiza ninguna escala.`);
+        } else {
+            console.log(`${flights[i].id} - El vuelo con origen ${flights[i].from} y destino ${flights[i].to} tiene un coste de ${flights[i].cost}€ y realiza escalas.`);
+        }
+    }
+}
+//Shows flights with scales
+function showScaleFlights(){
+    for(let i in scaleFlights){
+        
+        if (scaleFlights[i].scale == false) {
+            console.log(`${scaleFlights[i].id} - El vuelo con origen ${scaleFlights[i].from} y destino ${scaleFlights[i].to} tiene un coste de ${scaleFlights[i].price}€ y no realiza ninguna escala.`);
+        } else {
+            console.log(`${scaleFlights[i].id} - El vuelo con origen ${scaleFlights[i].from} y destino ${scaleFlights[i].to} tiene un coste de ${scaleFlights[i].price}€ y realiza escalas.`);
+        }
+    }
+}
+//Shows last 5 flights of the day
+function showLastFlights(){
+
+    lastFlights = [];
+    lastFlightsArray();
+
+    for(let i in lastFlights){
+        // Changes console log depending on scales number
+        if (lastFlights[i].scale == false){
+            console.log(`${lastFlights[i].id} - El vuelo con origen ${lastFlights[i].from} y destino ${lastFlights[i].to} tiene un coste de ${lastFlights[i].price}€ y no realiza ninguna escala.`);
+        } else {
+            console.log(`${lastFlights[i].id} - El vuelo con origen ${lastFlights[i].from} y destino ${lastFlights[i].to} tiene un coste de ${lastFlights[i].price}€ y realiza escalas.`);
+        }
+    }
+}
+
+/************ Small functionalities ************/
+
+//Says goodbye to the user
+function exit(){
+    alert("¡Hasta luego!");
+}
+
+
+/************ Main function ************/
+
+function airline(){
+
+    let userName = prompt(`Bienvenido a Skylab Arlines. Introduzca su nombre de usuario:`);
+
+    if(userName === null){

```

Comment:
Maybe an issue of project vague goals, but...:
Force user to register. 
If username === null show prompt "Bienvenido a Skylab Arlines. Introduzca su nombre de usuario:"